src/FieldDefinition.hack

```diff
@@ -34,4 +35,16 @@ final class FieldDefinition<TParent, TRet, TResolved> implements IFieldDefinitio
 
         return await $this->type->resolveAsync($value, $field, $vars);
     }
+
+    public function getName(): string {
+        return $this->name;
+    }
+
+    public function getType(): Introspection\__Type {
+        if ($this->type is Types\NullableInputType<_> || $this->type is Types\NullableOutputType<_, _>) {
+            return $this->type;
```

Comment:
I wonder if instead of returning `$this->type` here we can just check:

```
if ($this->type is Introspection\IHasInnerType) {
   return $this->type->getInnerType();
}
```

and then drop `IntrospectNullableType`...

Reply:
That would result in unwrapping lists, right? Which I don't think is intended, since it should be possible to return a "kind" which is a list.

Reply:
Yea, `IHasInnerType` would need to be renamed to make it only apply to nullables since those are the ones we just want to skip to the inner type. This would let us skip to the inner types for introspection but we still need to define `getName` and `getKind` on `NullableOutputType` and `NullableInputType` since they extend from `BaseType`. It seems weird to have to define those even though they'll never be called 🤔 .

The current approach seems easier to follow.

Reply:
You could also write a `getTypeForIntrospection()` method on `BaseType` and then just call it here. Then `NullableInputType` and `NullableOutputType` could override it to return `$this`, while the base implementation could return `new Introspection\NonNullableType($this->type)`.

 This is fine though.

Reply:
Also wondering if it might make sense for `getName()` and `getKind()` on nullable types to automatically forward to the inner type? Then instead of needing to unwrap these types, maybe you could just call these methods directly.

And finally, why are we returning `$this->type` if the type is nullable? Don't we want to return `$this->type->getInnerType()` since the nullable wrapper type shouldn't be surfaced by the introspection system?

Reply:
What if we had `getType()` and `getIntrospectionType()` as 2 different methods here? I haven't really thought it through very much but it feels like that might give us more type safety (e.g. we won't accidentally return a NonNullable-wrapped type during execution/validation; or accidentally return a Nullable-wrapped type for introspection)